### PR TITLE
Added a new UI for Web Client

### PR DESCRIPTION
**Pull request by onmax**
[Check the changes online](https://nimiq.github.io/developer-center/build/web-client/)

image

## Summary

- The auto-generated documentation has been moved to `build/web-client/reference`.
- The auto-generated `README.md` file is no longer visible in the Developer Center.
- I have added the following new pages
  - `index.md`: A quick list of features of the web client.
  - `getting-started.md`: A short description of Nimiq and a demo of how easy it is to connect to the network.
  - `installation.md`: A page with a list of the integrations we support. This concept of "integrations" may be a bit ambiguous, as we don't really have a separate support for each of them, but I think it's something very familiar to developers. I used this as a reference: [`TailwindCSS`](https://tailwindcss.com/docs/installation) and [`UnoCSS`](https://unocss.dev/guide/#integrations)

## Why do I think this is important?

With the release of Nimiq 2.0, we will have some new eyes on our project. Some of those eyes will be developers, and I think it is our responsibility to make it as easy as possible to create a new project and start developing with Nimiq.

I think the new documentation does just that: You can be up and running with Nimiq in less than a minute.

### CODE REVIEW

**Comment by onmax** [comment]
@sisou I don't want to share it yet with the team. First I would like to fix the deployment issue. Do you have access to the settings of the project?

See [this run](https://github.com/nimiq/developer-center/actions/runs/8893581683). Thanks

File: build/web-client/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+```js
+client.addHeadChangedListener((head) => {
+  console.log('New head:', head)
+})
+```
+
+## Issues, bugs and feedback
+
+This is an early version of the client code compiled on WebAssembly, and as such there may be issues and friction, especially now that more people are trying it in more environments than we could ever test ourselves.
```
Comment (Eligioo) [inline build/web-client/getting-started.md]:
```suggestion
This is an early version of the client code compiled to WebAssembly, and as such there may be issues and friction, especially now that more people are trying it in more environments than we could ever test ourselves.
```

File: build/web-client/integrations/nuxt.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Nimiq Web Client in Vite
```
Comment (Eligioo) [inline build/web-client/integrations/nuxt.md]: You are talking about Vite here but the file is called `nuxt.md`

**Review by Eligioo — APPROVED** [review]

File: build/web-client/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+next: ./installation/
+---
+
+# What's Nimiq?
+
+Nimiq is an open source crypto designed for simplicity and ease of use. With our browser-first approach, Nimiq works directly within web browsers without the need for additional software, allowing developers easy integration and users seamless interaction.
```
Comment (ines-toupeira) [inline build/web-client/getting-started.md]: ... and seamless user interaction.

File: build/web-client/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+The protocol is based on the [Albatross consensus algorithm](/learn/), a proof-of-stake system renowned for its speed, security and low power consumption. Albatross supports high transaction throughput, capable of handling thousands of transactions per second.
```
Comment (ines-toupeira) [inline build/web-client/getting-started.md]: ... and can handle thousands of transactions per second.

**Comment by Albermonte** [comment]
## Imports
I think having `@nimiq/core-web/web` for the imports is repetitive and will cause problems with new users, if I'm already importing `core-web` having to also select the `web` subfolder is not intuitive. Maybe changing the package name to simply `@nimiq/core` and then accessing the platform with subfolders like `@nimiq/core/web`, `@nimiq/core/node`, etc.
But this has probably more to do with how Core Team exports the packages.

## Installation
The order in the grid in `Installation` is not the same as the order in the `Integrations` dropdown.
1. Vite, 2. ESM, 3. Webpack...
image

1. CommonJS, 2. ESM, 3. NextJS...
image

## Package Manager
Any reason why `CommonJS` doesn't have the pnpm and bun options?
Also the JS version displays nothing
image

Also, if it could be possible to remember the selected package manager would be OP.

Same for open and closed dropdowns, right now the `Enums` and `Interfaces` are always open when I go to a new page, this is annoying when I'm looking into the `UI` part for example, having to scroll down the whole sidebar to see the `UI` sidebar navigation again. Focusing the scroll in the active `<li>` will be helpful too

**Comment by onmax** [comment]
@Albermonte:

> Imports

I completely agree with you and we have more than enough time to change this. @sisou what's your opinion? 

It is important to consider that this package will be with us for years, so discussing about this kind of details I think is valuable.

> CommonJS, 2. ESM, 3. NextJS...

You are right, but I like the order of the grid like that. The order of the sidebar comes from the file names of the markdown. Implementing our custom order requires multiple lines of code. Not sure if it is worth to do it. We'll see 🤔

> Package Manager

Error with markdown files. Good catch.

> Also, if it could be possible to remember the selected package manager would be OP.

I tried, but didn't manage to do something reliable. 

> Same for open and closed dropdowns

I will close them by default

File: .github/workflows/deploy.yaml [hunk 5]
```diff
@@ -4,7 +4,7 @@ on:
   # Runs on pushes targeting the `main` branch. Change this to `master` if you're
   # using the `master` branch as the default branch.
   push:
-    branches: [main]
+    branches: [onmax/new-web-client]
```
Comment (jsdanielh) [inline .github/workflows/deploy.yaml:7]: I guess this shouldn't be part of the PR

File: .github/workflows/deploy.yaml [hunk 6]
```diff
@@ -27,6 +27,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          ref: 'onmax/new-web-client'
```
Comment (jsdanielh) [inline .github/workflows/deploy.yaml:31]: Same here, why your branch?

File: build/web-client/getting-started.md (hunk 3 above)
Comment (jsdanielh) [inline build/web-client/getting-started.md]: I'd rephrase this as: providing easy integration for developers and seamless interaction for users.

File: build/web-client/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+```js
+client.addHeadChangedListener((head) => {
+  console.log('New head:', head)
+})
+```
+
+## Issues, bugs and feedback
+
+This is an early version of the client code compiled to WebAssembly, and as such there may be issues and friction, especially now that more people are trying it in more environments than we could ever test ourselves.
```
Comment (jsdanielh) [inline build/web-client/getting-started.md]:
```suggestion
This is an early version of the client code compiled to WebAssembly, and as such there might be issues and friction, especially now that more people are trying it in more environments than we could ever test ourselves.
```

File: build/web-client/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+# What's Nimiq?
+
+Nimiq is an open source crypto designed for simplicity and ease of use. With our browser-first approach, Nimiq works directly within web browsers without the need for additional software, allowing developers easy integration and users seamless interaction.
+
+The protocol is based on the [Albatross consensus algorithm](/learn/), a proof-of-stake system renowned for its speed, security and low power consumption. Albatross supports high transaction throughput, capable of handling thousands of transactions per second.
+
+## The Nimiq Web Client
+
+The Nimiq Web Client is a JavaScript library that allows you to interact with the Nimiq blockchain directly in your browser. It provides a simple interface to create wallets, send transactions and interact with the blockchain.
```
Comment (jsdanielh) [inline build/web-client/getting-started.md]:
```suggestion
The Nimiq Web Client is a JavaScript library that allows you to interact with the Nimiq PoS blockchain directly in your browser. It provides a simple interface to create wallets, send transactions and interact with the blockchain.
```

File: .github/workflows/deploy.yaml (hunk 5 above)
Comment (onmax) [reply .github/workflows/deploy.yaml:7]: nop, this is just temporarly. GitHub Pages does not allow multiple branches

**Review by Albermonte — APPROVED** [review]

**Review by jsdanielh — APPROVED** [review]

**Review by janinedotgm — CHANGES_REQUESTED** [review]
As I'm just beginning to familiarize myself with the project and the decisions that have shaped our current documentation, I recognize that there may be context I'm not yet aware of. From my initial navigation and understanding of the docs, I suggest considering the following thoughts:

## Overview Section
 
- The current content of the overview page doesn't seem to provide a clear overview. Typically, I would expect a brief description of what the web client is, something like the passage under "Getting Started": "The Nimiq Web Client is a JavaScript library that allows you to interact with the Nimiq PoS blockchain directly in your browser. It provides a simple interface to create wallets, send transactions and interact with the blockchain." Additionally, it might be helpful to briefly explain the relationship to the JSON RPC API. I think the web client is build on top of the RPC API, right? Maybe just mention that. Then we could set the button "Use RPC instead" in a context, with one sentence when they might need to use RPC or just link the word in the in the sentence where we explain the web client.
- The purpose of the content boxes on the page is currently unclear (to me). Are they intended to highlight benefits, reasons to use the web client, or important facts users need to be aware of? Clarifying this could enhance understanding and engagement.
- Is there a specific reason for hiding the left navigation bar and the table of contents when clicking on "Overview" in the Web Client Section? If it’s not a landing page, it might be less confusing to keep these elements visible. As it stands, navigating away and then having to use the back button to return can be disorienting.
- It would be beneficial if the headers in the left navigation bar were links or anchors to the sections. For instance, when I am on the "JSON-RPC Specification" tab and click on the heading "Web Client," I expect to land on the overview page of the web client.

## Getting Started Section

The headline "What's Nimiq?" initially confused me. I thought it might be a bug or a link pointing to the wrong page. Given that visitors are likely familiar with Nimiq by the time they reach this part of the documentation, linking the first mention of Nimiq to an introductory page might be better.

**Comment by onmax** [comment]
@jvvv-ship-it  thanks for your feedback 🙂

>  I would expect a brief description of what the web client is

yes, agree. I will add it.

> I think the web client is build on top of the RPC API, right?

Not really, they are two completely different ways of interacting with the blockchain. Web client is the client that most people will need.

RPC on the other hand is the way you have to communicate with a node (usually your private node), so if you are a validator in a server, it is an easy way to interact with your node. 

Anyway, I think it would be nice to have a comparison page between both clients. We will keep the discussion in #8.

>  Are they intended to highlight benefits, reasons to use the web client, or important facts users need to be aware of...

Yes, just features. Not very useful, but some nice content imo. I just copied the same structure as other projects using Vitepress:

- https://vueuse.org/
- https://vitepress.dev/
- https://unocss.dev/
- https://www.radix-vue.com/

> Is there a specific reason for hiding the left navigation bar and the table of contents when clicking on "Overview" in the Web Client Section?

I thought it looked better. The idea is that the Web Client page will be the first page that people go to when they google, rather than something they discover when they explore the Developer Center, and it looks better

> or instance, when I am on the "JSON-RPC Specification" tab and click on the heading "Web Client"

Yes, I will uupdate it

**Comment by sisou** [comment]
The NPM package naming comes from the PoW chain. First, we had `@nimiq/core`. It was the main package, and was only for NodeJS (because that's what the PoW client runs in for nodes). Then we added `@nimiq/core-web` for the light version of the client that runs in browsers and doesn't have NodeJS-specific stuff. When these packages came out, the ecosystem didn't have support for export maps and I don't remember what led to the decision to use a separate package for browsers (@jeffesquivels maybe has a better memory).

For the PoS client, we initially only had a browser version (what is now the `/web` export). So with @jeffesquivels we agreed to use the `@nimiq/core-web` package, so users don't have to change their imports. Since then I added the `/bundler` export (which is currently the default export for browsers, hence the need to specify the `/web` subpath when requiring the `web` target), but also the `/nodejs` export (which is automatically used in NodeJS environments, thanks to the export map).

I agree this has become confusing and bad DX. 

So I see two ways forward:
1. Continue using `@nimiq/core-web` for the builds targeted at browsers (`bundler` and `web` targets), but use the `@nimiq/core` package for the `nodejs` target (keeping both packages active).
2. Change over to the `@nimiq/core` package and deprecate `@nimiq/core-web`. We shouldn't have to change any default export, as the NodeJS environment should still load the `nodejs` target by default with the export map (needs verification).

**Comment by onmax** [comment]
Thanks for the detailed explanation. Let's talk about renaming the package in #9
